.github/workflows/deploy_ensnode_subgraph.yml

```
@@ -0,0 +1,107 @@
+name: Build and deploy ENSNode
+
+on:
+  push:
+    branches:
+      - feat/railway-ensnode-deployments
+      - main
+      - alpha
+      - subgraph
+
+jobs:
+  image-build-and-push:
+    runs-on: ubuntu-latest
+    outputs:
+      tag: ${{ steps.determine_tag.outputs.tag }}
+      svc_id: ${{ steps.determine_tag.outputs.svc_id }}
+      docker_image_id: ${{ steps.determine_tag.outputs.docker_image_id }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Extract Branch Name
+        run: echo "BRANCH=${GITHUB_REF#refs/heads/}" >> $GITHUB_ENV
+    
+      # We need to recognize which branch had push incomin
+      # Based on that create Docker image with corresponding label - use TAG variable for this
+      # For alhpa/subgraph TAG we need also to specify corresponding Railway SVC_ID which is used for service redeployment
+      - name: Determine tag
+        id: determine_tag
+        run: |
+            case "$BRANCH" in
+              "main")
+                echo "TAG=latest" >> $GITHUB_ENV
+                ;;
+              "alpha")
+                echo "TAG=alpha" >> $GITHUB_ENV
+                echo "SVC_ID=${{ secrets.ALPHA_SVC_ID }}" >> "$GITHUB_OUTPUT"
+                ;;
+              "subgraph")
+                echo "TAG=subgraph" >> $GITHUB_ENV
+                echo "SVC_ID=${{ secrets.SUBGRAPH_SVC_ID }}" >> "$GITHUB_OUTPUT"
+                ;;
+              "feat/railway-ensnode-deployments")
+                echo "TAG=subgraph" >> $GITHUB_ENV
+                echo "SVC_ID=${{ secrets.SUBGRAPH_SVC_ID }}" >> "$GITHUB_OUTPUT"
+                ;;
+              *)
+                echo "Branch not recognized, skipping workflow"
+                exit 1
+            esac
+            echo "TAG=$TAG" >> "$GITHUB_OUTPUT"
+
+      - name: Build & Push Docker image for the ENSNode
```

```suggestion
      - name: Build & Push Docker image for ENSNode to GitHub
```

Is that correct? That this is pushing the image to GitHub specifically?

---

They are calling it 'Github Container registry'. Adjusted